src/assets/scripts/client/trafficGenerator/SpawnPatternModel.js

```diff
@@ -0,0 +1,487 @@
+        //       the `spawnPattern` section of airport.json
+        /**
+         * Minimum milisecond elay between spawn.
+         *
+         * Is used as the lower bound when getting a random delay value.
+         *
+         * @property _minimumDelay
+         * @type {number}
+         * @default -1
+         * @private
+         */
+        this._minimumDelay = -1;
+
+        /**
+         * Lowest altitude an aircraft can spawn at
+         *
+         * @property _minimumAltitude
+         * @type {number}
+         * @default -1
+         * @private
+         */
+        this._minimumAltitude = -1;
+
+        /**
+         * Highest altitude an aircraft can spawn at
+         *
+         * @property _maximumAltitude
+         * @type {number}
+         * @default -1
+         * @private
+         */
+        this._maximumAltitude = -1;
+
+        /**
+         * Speed of spawning aircraft
+         *
+         * @property speed
+         * @type {number}
+         * @default 0
+         */
+        this.speed = 0;
+
+        /**
+         * Heading of a spawnning aircraft
+         *
+         * @property heading
+         * @type {number}
+         * @default -1
+         */
+        this.heading = -1;
+
+        /**
+         *
+         *
+         * @property position
+         * @type {array}
+         * @default []
+         */
+        this.position = [];
+
+        /**
+         * List of possible airlines a spawning aircraft can belong to.
+         *
+         * @property airlines
+         * @type {array<string>}
+         * @default []
+         */
+        this.airlines = [];
+
+        /**
+         * List of airlines enumerated by weight
+         *
+         * In english, if the value of `this.airlines` was:
+         * ```
+         * [
+         *     ['aal', 5],
+         *     ['ual', 2]
+         * ]
+         * ```
+         * This property would have a length of 7, with 5 entires of `aal` and two entries of `ual`.
+         * The reason for this is to provide an easy way to find a weighted value. Now all we need is
+         * a random index and the value located at that index.
+         *
+         * @property _weightedAirlineList
+         * @type {array}
+         * @default []
+         */
+        this._weightedAirlineList = [];
+
+        this.init(spawnPatternJson, navigationLibrary);
+    }
+
+    /**
+     * List of airline icaos for this spawnPattern
+     *
+     * @property airlineList
+     * @return {array<string>}
+     */
+    get airlineList() {
+        return _map(this.airlines, (airline) => airline.name);
+    }
+
+    /**
+     * A number representing the initial altitude of a spawning aircraft
+     *
+     * @property altitude
+     * @return {number}
+     */
+    get altitude() {
+        // this might not need to be random within a range
+        return _random(this._minimumAltitude, this._maximumAltitude);
```

Comment:
`_random` should only return a whole number between the min and max. so that shouldn't be a thing, unless you saw that to be a thing?  In which case I need to write a test for it 😉